src/util/fs_helpers.h

```diff
@@ -62,6 +62,19 @@ void ReleaseDirectoryLocks();
 bool TryCreateDirectories(const fs::path& p);
 fs::path GetDefaultDataDir();
 
+std::string PermsToString(fs::perms p);
+
+struct CookiePerms {
+    struct Perm {
+        const char* name;
+        fs::perms permissions;
+    };
+
+    static constexpr Perm owner  {"owner",  fs::perms::owner_read | fs::perms::owner_write};
+    static constexpr Perm group  {"group",  owner.permissions     | fs::perms::group_read};
+    static constexpr Perm all    {"all",    group.permissions     | fs::perms::others_read};
+};
```

Comment:
This feels overengineered. Why can't we stick to fs::perms internally and just map the string in a function?

Reply:
Now simplified in d8d47ab395b8fa341e0aac2fe01a4b9488af5347